sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -214,50 +220,54 @@ private Response<?> receiveResponse(HttpRequest httpRequest, HttpURLConnection c
                 throw LOGGER.logThrowableAsError(new RuntimeException(NO_LISTENER_ERROR_MESSAGE));
             }
 
-            processTextEventStream(this, httpRequest, connection.getInputStream(), listener);
-        } else {
-            RequestOptions options = httpRequest.getRequestOptions();
-            ResponseBodyMode responseBodyMode = null;
+            serverSentResult = processTextEventStream(connection.getInputStream(), listener);
 
-            if (options != null) {
-                responseBodyMode = options.getResponseBodyMode();
+            if (Thread.currentThread().isInterrupted() || !retryReconnect(serverSentResult, httpRequest)) {
+                listener.onError(serverSentResult.getException());
+            } else {
+                this.send(httpRequest).close();
```

Comment:
I don't fully understand the logic here. If we're not retrying the connection we emit an `onError`, but wait if we consumed the connection successfully?

If we're retrying, why do we eagerly close the new request? Shouldn't that be what is returned instead of the initial request? Also, I've realized this is bypassing the whole retry policy mechanism, shouldn't we be honoring that?

Reply:
>I don't fully understand the logic here. If we're not retrying the connection we emit an onError, but wait if we consumed the connection successfully?

If the current thread has been interrupted or if the default reconnect method returns false, it means an error has occurred. And so we emit an `onError`.